packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/utils.tsx

```diff
@@ -98,6 +98,26 @@ export function spanContainsErredSpan(spans: Span[], parentSpanIndex: number) {
   return false;
 }
 
+/**
+ * Collect the spanIDs of all descendant spans that contain an error tag.
+ * Descendants are spans after parentSpanIndex with strictly greater depth
+ * until depth drops back to parent's depth or less.
+ */
+export function getDescendantErroredSpanIDs(spans: Span[], parentSpanIndex: number): string[] {
```

Comment:
it seems `isErrorSpan` function already performs the necessary calculation, there is no need to redo it (and slowing down rendering). We just need to change the function to return more than a boolean, like isErrorSpan(): {isError: bool, selfError: bool}`